cgp/genome.py

```diff
@@ -270,6 +269,105 @@ def randomize(self, rng: np.random.RandomState) -> None:
         # accept generated dna if it is valid
         self.dna = dna
 
+    def splice_dna(self, dna_insert: List[int], internal_start_node: int = 0) -> int:
+        """Splice the dna.
+
+        Replaces a segment of the dna starting from internal start node with the dna insert
+
+        Parameters
+        ----------
+        dna_insert: List[int]
+            dna segment to be inserted.
+        internal_start_node: int
+            idx of the internal node, where the insert starts.
+            Defaults to 0.
+
+        Returns
+        ----------
+        address_last_inserted_node: int
+            address of the last node inserted.
+        """
+        if len(dna_insert) % self._length_per_region != 0 or len(dna_insert) == 0:
+            raise ValueError("Length of inserted dna must be a multiple of region length")
+
+        n_inserted_nodes = len(dna_insert) // self._length_per_region
+
+        if internal_start_node + n_inserted_nodes >= self._n_hidden:
+            raise ValueError("Inserted dna too long")
+
+        dna = self.dna
+
+        start_idx = (self._n_inputs + internal_start_node) * self._length_per_region
+        end_idx = start_idx + len(dna_insert)
+        dna[start_idx:end_idx] = dna_insert
+
+        self.dna = dna
+
+        address_last_inserted_node = self._n_inputs + internal_start_node + n_inserted_nodes - 1
+
+        return address_last_inserted_node
+
+    def change_output_address(self, address_node: int, output_node_idx: int = 0):
+        """Change the address of an output gene
+
+        Parameters
+        ----------
+        address_node: int
+            Address of the node from which the output should read
+        output_node_idx: int
+            If defined, idx of the output node for which the address node is changed.
+            Defaults to 0.
+
+        Returns
+        ----------
+        None
+        """
+        if address_node < 0 or address_node > self._n_hidden:
+            raise ValueError("Address node non valid")
+        if output_node_idx < 0:
+            raise ValueError("Output node idx must be non-negative")
+        if output_node_idx > self._n_outputs - 1:
+            raise ValueError("Output node idx must be smaller than number of outputs")
+
+        dna = self.dna
+
+        idx_output_address_gene = (
+            self._n_inputs + self._n_hidden + output_node_idx
+        ) * self._length_per_region + 1  # address gene has offset 1
+        dna[idx_output_address_gene] = address_node
+        self.dna = dna
+
+    def set_prior(
```

Comment:
if not sure whether "prior" is the best name. can't we just say "set_expression_for_output" or something like that?